active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+            <Border Background="Red" Margin="3" CornerRadius="10" Width="20" Height="20" VerticalAlignment="Top">
+                <TextBlock Text="!" Foreground="White"
+                HorizontalAlignment="Center" VerticalAlignment="Center"/>
+            </Border>
+            <TextBlock Text="{x:Bind (Validation.Errors)[0].ErrorContent, Mode=OneWay}"
+        </StackPanel>
+    </DataTemplate>
+</ResourceDictionary>
+
+<TextBox Validation.ErrorTemplate="{StaticResource ValidationErrorTemplate}"/>
+```
+
+## Change the resource color for the error and description messages
+Colors for the error and description messages are defined in a standard color/brush pattern in generic.xaml that can be overridden at an app level.
+
+```xaml
+<Application.Resources>
+
+    <!--Color for error messages-->
+    <Color x:Key="SystemErrorColor">#CE3880</Color>
+    <SolidColorBrush x:Key="SystemControlErrorTextForegroundBrush" Value="SystemErrorColor"/>
+
+    <!--Brush for description messages-->
+    <SolidColorBrush x:Key="SystemControlDescriptionTextForegroundBrush" Value="SystemControlPageTextBaseMediumBrush"/>
+
+</Application.Resources>
+```
+
+## Change description text for validation
+
+Description text provides input to the end user on what format or content is expected from the control when it is not clear from the header and placeholder text.
+
+Developers can choose to either show the description text for a control or not. Default is that description text is off. 
+
+Description text should be used to complement headers and placeholders. Guidance on what each corresponds to is as follows: 
+1.	Headers give you an indication of what the control represents. It is used to catch the eye of a user scanning your form or to refer to a field.
+2.	Placeholders provide a succinct description of what should be input in the box. However, they aren’t an ideal pattern for guiding users input because user testing continually shows that placeholders in form fields often hurt usability more than help it. This is because: 
+    - Disappearing placeholder text strains users’ short-term memory.
+    -	Users may mistake a placeholder for data that was automatically filled in.
+    -	The default light-grey color of placeholder text has poor color contrast against most backgrounds.
+    -	Not all screen readers read placeholder text aloud
+    -	Often incompatible with autofill – you need to delete field content before finding the hint.
+3.	Description text is used to complement placeholder text. It provides more detailed formatting guidance that is needed while editing the field. 
+
+#### XAML
+```xaml
+    <TextBox 
+            x:Name="FirstName" 
+            Header="First Name:" 
+            Text="{x:Bind   ViewModel.Person.FirstName, 
+                            UpdateSourceTrigger=LostFocus, 
+                            Mode=TwoWay}"
+            DescriptionText="Use alphabetical characters (a-Z)."
+        />
+```
+
+#### C# (Model)
+```csharp
```

Comment:
Is this meant to work with the XAML directly above or an example of the C# way of doing the same thing?
That the `DescriptionText` in the XAML and the `ErrorMessage` in the C# use the same text is potentially confusing. Are developers expected to have this duplication?

Reply:
The idea here was that the Description text will be replaced with the error text, but this does assume that `InputValidationKind` is set to `Inline`. When using `Compact` i'm pretty sure the `Desription` will still show

Reply:
This answers some of the additional comments I've just added above. Thanks for the clarification.